### PR TITLE
Add chart from upstream

### PR DESCRIPTION
**Pull request by pipo02mix**
Following our internal process to add the upstream chart as a subtree

### CODE REVIEW

File: helm/goldilocks-app/Chart.yaml [hunk 1]
```diff
@@ -1,11 +1,10 @@
 apiVersion: v2
-appVersion: "v4.4.0"
-version: "[[ .Version ]]"
+appVersion: "v4.8.3"
+version: 6.7.0
```
Comment (pipo02mix) [inline helm/goldilocks-app/Chart.yaml:3]:
```suggestion
version: "[[ .Version ]]"
```

**Comment by hervenicol** [comment]
Wasn't it already a subtree?
In Atlas, we tend to migrate to chart dependency.

One of the reasons is it's hard to know our code from upstream code when reading this kind of PR.
The drawback of migrating to chart dependency is it's changing the values schema (adding one level in the values tree).

File: helm/goldilocks-app/values.yaml [hunk 2]
```diff
@@ -14,11 +14,10 @@ metrics-server:
     create: true
 
 image:
-  registry: quay.io
   # image.repository -- Repository for the goldilocks image
-  repository: giantswarm/goldilocks
+  repository: us-docker.pkg.dev/fairwinds-ops/oss/goldilocks
```
Comment (hervenicol) [inline helm/goldilocks-app/values.yaml:18]: I think we should use a retagged image here.

File: helm/goldilocks-app/values.yaml [hunk 3]
```diff
@@ -14,11 +14,10 @@ metrics-server:
     create: true
 
 image:
-  registry: quay.io
```
Comment (hervenicol) [inline helm/goldilocks-app/values.yaml:17]: Did we lose the possibility to configure registry this way?
This is a standard value that is overridden for Chinese installations, if we don't have it we will need some special config for this app.

**Comment by pipo02mix** [comment]
> Wasn't it already a subtree?
  In Atlas, we tend to migrate to chart dependency.
  
  Do we have documentation for that? I was using intranet docs for upstream tracking

File: helm/goldilocks-app/values.yaml (hunk 3 above)
Comment (pipo02mix) [reply helm/goldilocks-app/values.yaml:17]: well they have changed from registry to repository. So the solution is to revert this change manually? I am not used to get upstream changes

**Comment by hervenicol** [comment]
> > Wasn't it already a subtree?
> > In Atlas, we tend to migrate to chart dependency.
> 
> Do we have documentation for that? I was using intranet docs for upstream tracking

Actually yes, we have! :tada: 
https://intranet.giantswarm.io/docs/dev-and-releng/how-to-migrate-to-chart-dependency/

Although you may be the first one reading it when actually needing it.
So feedback is very welcome!

File: helm/goldilocks-app/values.yaml (hunk 3 above)
Comment (pipo02mix) [reply helm/goldilocks-app/values.yaml:17]: should I include it then in the repo name? WDYT?

**Comment by pipo02mix** [comment]
This is ready to be merged. Just the question of the registry? what do we do when it is embedded in the repo name?

**Comment by hervenicol** [comment]
> Just the question of the registry? what do we do when it is embedded in the repo name?

This is the main problem when using upstream charts as dependency: we can't override everything we want. So we need to work upstream us much as possible.

The registry value is quite standard, should not be a problem to contribute the change upstream.
Question is: how critical is it to have it before merging this PR?

**Comment by pipo02mix** [comment]
> Question is: how critical is it to have it before merging this PR?

I do think we can merge it since it is an optional app and right now it is not running in China

**Review by hervenicol — APPROVED** [review]
Nice job, LGTM.

**Pull request merged**